File: src/sdk-reference/csharp/1/kuzzle/query/snippets/query.cs
try {
  KuzzleRequest request = new KuzzleRequest();
  request.controller = "document";
  request.action = "create";
  request.index = "nyc-open-data";
  request.collection = "yellow-taxi";
  request.id = "my-custom-document-id";
  request.body = "{\"trip_distance\": 4.23, \"passenger_count\": 2}";

  QueryOptions options = new QueryOptions();
  options.refresh = "wait_for";

  KuzzleResponse response = kuzzle.query(request, options);

  Console.WriteLine("Document created");
}
catch (UnauthorizedException e) {
  Console.Error.WriteLine("You are not connected");
}
catch (KuzzleException e) {
  Console.Error.WriteLine(e.getMessage());
}

File: src/sdk-reference/cpp/1/kuzzle/query/snippets/query.cpp
try {
  kuzzleio::kuzzle_request request;
  request.controller = "document";
  request.action = "create";
  request.index = "nyc-open-data";
  request.collection = "yellow-taxi";
  request.id = "my-custom-document-id";
  request.body = "{\"trip_distance\": 4.23, \"passenger_count\": 2}";

  kuzzleio::query_options options;
  options.refresh = "wait_for";

  kuzzleio::kuzzle_response* response = kuzzle->query(request, options);

  std::cout << "Document created" << std::endl;
}
catch (kuzzleio::UnauthorizedException& e) {
  std::cerr << "You are not connected" << std::endl;
}
catch (kuzzleio::KuzzleException& e) {
  std::cerr << e.what() << std::endl;
}
